# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../xmlrpc-c/xmlrpc-c-1.16.41-curl-headers.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Fix to not include deprecated and unneeded curl headers

a) curl/types.h

   The use of including <curl/types.h> was deprecated more than 7 years ago,
   and the header file was finally removed with the release of curl 7.21.7.

   Reference how upstream changed their own example code:
   https://github.com/bagder/curl/commit/ac28971aa61d28e5dd54888e34e958d1c742b461

b) curl/easy.h and curl/multi.h

   both header files are included anyway when including curl/curl.h thus it
   is not really needed to include them explicitely

--- a/lib/curl_transport/xmlrpc_curl_transport.c.orig	2012-05-12 20:31:25.439481606 +0200
+++ b/lib/curl_transport/xmlrpc_curl_transport.c	2012-05-12 20:37:28.806931349 +0200
@@ -83,9 +83,6 @@
 #include "xmlrpc-c/time_int.h"
 
 #include <curl/curl.h>
-#include <curl/types.h>
-#include <curl/easy.h>
-#include <curl/multi.h>
 
 #include "lock.h"
 #include "lock_pthread.h"
--- a/lib/curl_transport/curltransaction.c.orig	2012-05-12 20:38:58.849650689 +0200
+++ b/lib/curl_transport/curltransaction.c	2012-05-12 20:42:28.326819943 +0200
@@ -15,8 +15,6 @@
 #include "version.h"
 
 #include <curl/curl.h>
-#include <curl/types.h>
-#include <curl/easy.h>
 
 #include "curlversion.h"
 
--- a/lib/curl_transport/curlmulti.c.orig	2012-05-12 20:42:56.422179313 +0200
+++ b/lib/curl_transport/curlmulti.c	2012-05-12 20:43:09.567824771 +0200
@@ -17,9 +17,6 @@
 #endif
 
 #include <curl/curl.h>
-#include <curl/types.h>
-#include <curl/easy.h>
-#include <curl/multi.h>
 
 #include "mallocvar.h"
 #include "xmlrpc-c/util.h"
